### ISSUE TITLE
1000 lines might be to few

### ISSUE THREAD

**Issue by Gaweringo** [jkavalik/GcodeThumbnailExtension#3]
The Extension works well, with preview images of 16x16 (1st Image) and 128x128 (second image), but when the second image gets bigger (256x256) the image information goes past the 1000 lines, and thus no preview is shown.
Maybe in this case the first image could be shown or just increase the searched through lines (might be to slow).
But I don't know much about Shell Extensions.

(Nice work btw appreciate it :)

**Comment by jkavalik** [comment]
Hello, thank you for both the report and the praise.
Could you please give me an example gcode file where you get such problems?

I am checking now a freshly generated file and the second thumbnail is 220x124 and spans something like 150 lines, with an image about double the size I would expect that to also be about double, say 250-350 lines, not over 1000. Still that limit may be too conservative as I used the 220x124 as a reference because my slicer does not seem to generate bigger ones. I do not want to search the entire file because that would lead to wasting reading many megabytes from possibly an external media (flash card/drive), but I am not against using 2k or 10k if it helps.

Another weird thing is that if the parsing of the second image fails the first should be used at least, and that the 1k limit should actually only matter when finding the beginning of the image data and should not truncate the actual image anyway, but I might have made a mistake somewhere :)

**Comment by Gaweringo** [comment]
Ah, looks like I reacted too fast.
I had just installed the extension and tried a file with a preview of 16x16 and 256x256. This file to be exact:
[Dual_airduct_-_raised_version.zip](https://github.com/jkavalik/GcodeThumbnailExtension/files/6176072/Dual_airduct_-_raised_version.zip)
And I got no preview, so I looked at the file and saw, that the thumbnail data went over the 1000th line. I then reduced the image to 128x128 and it worked (didn't reach the 1000th line) and so I thought there was a problem with the extension.
But it looks like it might have just been, that the extension wasn't really running yet, as when I now try this file it works as intended, showing the 256x256 preview. Even bigger previews like 512x512 work.

Should have double-checked the problem before creating the issue.
But hey, at least it pushed me to thank you for your work :)
